src/Pipeline.cpp

```diff
@@ -555,6 +566,21 @@ Target Pipeline::get_compiled_jit_target() const {
 }
 
 void Pipeline::compile_jit(const Target &target_arg) {
+    std::string debug_serialization = get_env_variable("HL_DEBUG_SERIALIZATION");
+    Pipeline deserialized_pipe;
+    if (!debug_serialization.empty()) {
+        std::string filename = generate_function_name() + ".hlpipe";
+        debug(0) << "intercept compile_jit, file: " << filename << "\n";
+        std::unordered_map<std::string, Internal::Parameter> call_params;
+        serialize_pipeline(*this, filename, call_params);
+        debug(0) << "intercept compile_jit, serialization completed\n";
+        deserialized_pipe = deserialize_pipeline(filename, call_params);
+        debug(0) << "intercept compile_jit, deserialization completed\n";
+        // deserialized_pipe.jit_handlers() = this->jit_handlers();
```

Comment:
As always, commented out code should be removed or have a comment, even for a draft PR :-)

(eg even just `// TODO: commented out pending foo`)